.travis/script.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+case "$TRAVIS_OS_NAME" in
+"linux")
+    sed -i.bak 's/openjdk:8/openjdk:'${JAVA}'/' share/docker/Dockerfile
```

Comment:
Can you change the single quotes to double quotes?

Reply:
Is there any way to leverage docker arguments rather than mutating the dockerfile itself? https://docs.docker.com/engine/reference/builder/#arg

Reply:
@kojiromike, that looks cool, I haven't seen that feature before. I'm going to dig in to see if that will do the trick for us. If for some reason it doesn't work out, @Fokko , I will replace all the single quotes on this line with double quotes.

FYI, the contents of this case statement came from our current [`.travis.yml` here](https://github.com/apache/avro/blob/ee4c2aaf4a5a0e1108dc01b157150d4629dd7881/.travis.yml#L39).

Reply:
So close! Docker supports this scenario, but it doesn't look like Yetus does. Yetus doesn't accept and pass along Dockers `--build-arg` option as far as I can tell. See [Yetus' Docker documentation here](https://yetus.apache.org/documentation/in-progress/precommit-docker/) and the `test-patch` command line help below.

```
Docker options:
--docker-bash-debug=<bool>      Enable bash -x mode running in a container (default: false)
--docker-cache-from=<image>     Comma delimited images to use as a cache when building
--dockercmd=<file>              Command to use as docker executable (default: '/usr/bin/docker')
--dockerdelrep                  In Docker mode, only report image/container deletions, not act on them
--dockerfile=<file>             Dockerfile fragment to use as the base (default:
                                '/usr/lib/precommit/test-patch-docker/Dockerfile')
--dockerind=<bool>              Enable Docker-in-Docker by mounting the Docker socket in the container (default:
                                'false')
--dockermemlimit=<num>          Limit a Docker container's memory usage (default: 4g)
--docker-platform=<plat>        Use a platform string for building and pulling (default: )
--dockerprivd=<bool>            Run docker in privileged mode (default: 'false')
--docker-socket=<socket>        Mount given socket as /var/run/docker.sock into the container when Docker-in-Docker
                                mode is enabled (default: '/var/run/docker.sock')
--docker                        Spawn a docker container
--docker-tag=<tag>              Use the given Docker tag as the base
--docker-work-dir=<directory>   Disposable, safe dir for precommit work (default: /precommit
```

Reply:
That would be a nicer way of doing it, but that is outside of the scope of this PR.